src/main/java/com/gw2auth/oauth2/server/service/OAuth2TokenCustomizerService.java

```diff
@@ -150,49 +157,241 @@ private void customize(JwtEncodingContext ctx, String clientAuthorizationId, UUI
+                        gw2AccountForJWT.put("id", gw2AccountId.toString());
+
+                        if (scopes.contains(OAuth2Scope.GW2ACC_NAME)) {
+                            final String gw2AccountName = account.gw2AccountName();
+                            gw2AccountForJWT.put("name", gw2AccountName);
+                            gw2AccountLogging.log(String.format("Including Name %s", gw2AccountName));
+                        }
+
+                        if (scopes.contains(OAuth2Scope.GW2ACC_DISPLAY_NAME)) {
+                            final String displayName = account.displayName();
+                            gw2AccountForJWT.put("display_name", displayName);
+                            gw2AccountLogging.log(String.format("Including Displayname %s", displayName));
+                        }
+
+                        if (scopes.contains(OAuth2Scope.GW2ACC_VERIFIED)) {
+                            final boolean isVerified = verifiedGw2AccountIds.contains(gw2AccountId);
+                            gw2AccountForJWT.put("verified", isVerified);
+                            gw2AccountLogging.log(String.format("Including verified=%s to the JWT", isVerified));
+                        }
+
+                        if (!gw2ApiPermissions.isEmpty()) {
+                            final String subToken = subTokensByGw2AccountId.get(gw2AccountId);
+
+                            if (subToken != null) {
+                                gw2AccountForJWT.put("token", subToken);
+                                gw2AccountLogging.log("Added subtoken to the JWT");
+                            } else {
+                                gw2AccountForJWT.put("error", "Failed to obtain new subtoken");
+                                gw2AccountLogging.log("No subtoken available to add to the JWT");
+                            }
+                        }
+
+                        gw2AccountsForJWT.add(gw2AccountForJWT);
+                    }
+                }
+            }
+
+            ctx.getClaims().claim("gw2_accounts", gw2AccountsForJWT);
+        }
+
+        if (scopes.contains(OAuth2Scope.ID)) {
+            ctx.getClaims().subject(userAccountSub.toString());
+            logging.log("Including subject");
+        } else {
+            final byte[] buf = ByteBuffer.allocate(Long.BYTES * 3)
+                    .putLong(userAccountSub.getLeastSignificantBits())
+                    .putLong(userAccountSub.getMostSignificantBits())
+                    .putLong(ThreadLocalRandom.current().nextLong())
+                    .array();
+
+            final UUID sub = UUID.nameUUIDFromBytes(buf);
+            ctx.getClaims().subject("RETRACTED-" + sub);
+
+            logging.log(String.format("Not including subject: id scope not requested; including %s", sub));
+        }
+    }
+
+    private Pair<Instant, Map<UUID, String>> getSubTokens(UUID userAccountId,
```

Comment:
Just to quickly mention it: I'm not criticizing the use of `Pair` here because the tuple is not passed across class boundaries but is only used fairly locally (in this case, within the class).

Reply:
This one is really limited in scope, I didnt want to introduce a new record just for this